pkg/scheduler/flavorassigner/flavorassigner.go

```diff
@@ -441,6 +442,20 @@ func (a *FlavorAssigner) findFlavorForPodSetResource(
 			status.append(fmt.Sprintf("flavor %s not found", fName))
 			continue
 		}
+		if features.Enabled(features.TopologyAwareScheduling) {
+			// For PodSets which require TAS skip resource flavors which don't support it
+			if ps.TopologyRequest != nil && flavor.Spec.TopologyName == nil {
+				log.Error(nil, "Flavor does not support TopologyAwareScheduling", "Flavor", fName)
+				status.append(fmt.Sprintf("flavor %s does not support TopologyAwareScheduling", fName))
+				continue
+			}
+			// For PodSets which don't use TAS skip resource flavors which are only for TAS
+			if ps.TopologyRequest == nil && flavor.Spec.TopologyName != nil {
+				log.Error(nil, "Flavor supports only TopologyAwareScheduling", "Flavor", fName)
+				status.append(fmt.Sprintf("flavor %s supports only TopologyAwareScheduling", fName))
+				continue
+			}
+		}
```

Comment:
Shouldn't we move these logic to the `pkg/scheduler/flavorassigner/tas_flavorassigner.go` same as the `func assignTopology()`?
Let's separate the responsibility.

Reply:
We could move this easily to a helper function Iin tas_flavorassigner,  checkPodSetAndFlavorCompatible

However, I do the logic here to prevent trying to compute the assignments.

For example if PodSet is requesting TAS there is no point to compute assignment for non TAS flavors, so we can skip. Maybe we could compute the assignments and then drop, but I think it is more complicated.

So I don't see a clearly better place to inject with the logic.

Reply:
I just imagined the below code. My objective is to distribute the TAS-related codes to tas_flavorassigner.go.
Note that the below code is pseudo-code.

```go
func isFlavorAndAssignmentSupportTAS(ps, flavor, fName, status) bool { 
		// For PodSets which require TAS skip resource flavors which don't support it
		if ps.TopologyRequest != nil && flavor.Spec.TopologyName == nil {
			log.Error(nil, "Flavor does not support TopologyAwareScheduling", "Flavor", fName)
			status.append(fmt.Sprintf("flavor %s does not support TopologyAwareScheduling", fName))
			continue
		}
		// For PodSets which don't use TAS skip resource flavors which are only for TAS
		if ps.TopologyRequest == nil && flavor.Spec.TopologyName != nil {
			log.Error(nil, "Flavor supports only TopologyAwareScheduling", "Flavor", fName)
			status.append(fmt.Sprintf("flavor %s supports only TopologyAwareScheduling", fName))
			continue
		}
}
```

Reply:
sure, opened: https://github.com/kubernetes-sigs/kueue/pull/3391